pkg/cli/init_project.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/kubebuilder/cmd/util"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+
+	"github.com/spf13/cobra"
+)
+
+func (c CLI) newInitProjectCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initialize a new project",
+		Long: `Initialize a new project.
+
+For further help about a specific project version, set --project-version.
+`,
+		Run:     func(cmd *cobra.Command, args []string) {},
+		Example: c.getInitHelpExamples(),
+	}
+
+	// Register --project-version on the dynamically created command
+	// so that it shows up in help and does not cause a parse error.
+	cmd.Flags().String("project-version", c.defProjVersion, fmt.Sprintf("project version, (%s)", strings.Join(c.getAvailableProjectVersions(), ", ")))
+
+	// Pre-parse the project version and help flags so that we can
+	// dynamically bind to a plugin's init implementation (or not).
+	projectVersion, isHelpOnly := c.getBaseFlags()
+
+	// If only the help flag was set, return the command as is.
+	if isHelpOnly {
+		return cmd
```

Comment:
doesn't this mean that a plugin doesn't get to modify help?

Reply:
The intent with this was to handle `kubebuilder init --help`, in which case we could print out a general help. In this case, we don't know which plugin to use for help output.

Reply:
Does this have something to do with the use of `log.Fatal` where you were using `cdmErr` and `return` in create api and webhook commands?